### ISSUE TITLE
Drop hestiaRS and opting hestiaZIG?

### ISSUE THREAD

**Issue by hollowaykeanho** [ZORALab/Hestia#100]
# Question
```
Please state your question in a brief and ended with question mark (?). You
may include sub-questions with "if cases" to communicate more effectively
instead of many 'ping-pong' passing.
```
Given the extreme authoritharian goverence from Rust upstream in terms of tools, conventions, standards, naming conventions, and worst: incomplete support of cross-platform while spending time messing with logo and brand trademark, I really do not wish to continue supporting Rust in Hestia.

Given that Firefox (Gecko Engine) has been playing catch up, I'm very permisstic about supporting that language given the track records were too vocal, too little actions).

The few reasons I selected Rust initially were primarily because:

1. WASM
2. Embedded Microcontroller
3. Non-GC requirements

and their offered RAII. Since we're going to rebuild a standard library, RAII is not really a priority (we're going to deal with unsafe interfacing with hardware anyway).

I'm planning to replace with Zig, and Zig also has:

1. WASM
2. Quite friendly with embedded microtroller
3. Non-GC requirement
4. A lot better cross-compilation supports similar to Go

Both still failed to have built-in code coverage tool like Go did but it does not really matter since the libraries are inter-operable so Go can cover it.

@corygalyna , I need your input.

# Background
```
Please provide a short description of what you are trying to ask.
```

# Current Research Artifacts (COMPULSORY)
```
Please list out all your research artifacts that you did (e.g. link, blog post,
documentations, and etc) for us to gauge your context and current efforts. We
may not reply and will close the ticket with perjuidice if this section remains
empty and unconvincing.
```
1. https://en.wikipedia.org/wiki/Zig_(programming_language)
2. https://ziglearn.org/
3. https://ziglang.org/learn/
4. https://github.com/C-BJ/Awesome-Zig

# Attachment
```
Please drag and drop the necessary data files (e.g. screenshot, logs, etc)
below.
```

**Comment by corygalyna** [comment]
Community seems ok: https://github.com/ziglang/zig
Project is active.

The downside I can see is Rust being WASM first-class citizen while Zig can have some uphill battle. I do agree that the noise and nuisance is very de-motivating.

**Comment by corygalyna** [comment]
Give me time to look through. After all, you're still working on Phase 1.2.

**Comment by hollowaykeanho** [comment]
Yes Go stays. It's a matter of choosing a statically-linked non-GC language to compliment Go where Go can't reach.
